### DISCUSSION TITLE
How to get the text from the relationship picker?

### DISCUSSION THREAD

**Discussion by dpkg-i-foo-deb** [Budibase/budibase#4688] [Help]
For example, I have a 'beds' table and another one called 'statuses' who has a relationship to the beds table to tell it's status day by day... One bed has multiple statuses day by day

![Screenshot from 2022-02-24 22-07-30](https://user-images.githubusercontent.com/80849765/155646217-b5a1f242-5a2b-40e9-8f60-331f0799831f.png)

![Screenshot from 2022-02-24 22-12-43](https://user-images.githubusercontent.com/80849765/155646776-9123b9ea-ea37-45a1-ae2e-4c6dd7d5cddf.png)

But then, if I want to create a relationship to the 'statuses' table and select a display value, I cannot use the relationship... So I'm struggling in how to get it... I wondered if it was possible to get it from the relationship picker but no luck... I don't get the text but the ID...

![Screenshot from 2022-02-24 22-10-19](https://user-images.githubusercontent.com/80849765/155646525-8655acb3-8a50-4b2c-9e65-2b07b10f9f8e.png)

Sooo right now I'm kinda forcing the imaginary user to type the bed name by hand to save it to a row and select that row as the display one

![Screenshot from 2022-02-24 22-11-42](https://user-images.githubusercontent.com/80849765/155646682-20c06fcb-66d7-4165-9725-abffba468bd1.png)

And yeah that is the least usable thing you could ever imagine but I see no other way T_T

i don't know either if it makes sense or if it is possible. I'm a novice on this things... Thanks T_T

**Comment by mjashanks** [comment]
Hey @dpkg-i-foo-deb 

I'm not totally sure what you mean, but I'll have a go...

In my example below, I do the following

1. A form
2. I have a Data Provider, which loads from a table called "Brands"
3. Inside, an options picker, which loads its values from the Data Provider. I set it's "Field" to "selectedBrand"
4. Inside the Data Provider, a pararagh function with Javascript binding. The Binding searches the data in the Data Provier for the `selectedBrand`, then returns the `Name`

It looks like this:

![image](https://user-images.githubusercontent.com/3524181/155758165-bba322f1-b6f0-4dbd-83a2-0abcfde65e91.png)

My component structure:

```
> Form
  > Data Provider : "Brands" table
     > Options Picker : Loading from above data provider
     >  Paragraph: with the javascript binding below 
```

and javascript...

```javascript
const selectedBrand = $("All Brands.Rows").find(r => r._id === $("New Form.Fields.selectedBrand"));

return `Selected brand is: ${selectedBrand.Name}`;
```
